components/de_amp/browser/test/de_amp_browsertest.cc

```diff
@@ -0,0 +1,274 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/bind.h"
+#include "base/files/file_path.h"
+#include "base/path_service.h"
+#include "brave/components/de_amp/common/features.h"
+#include "brave/components/de_amp/pref_names.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/prefs/testing_pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "net/dns/mock_host_resolver.h"
+#include "third_party/re2/src/re2/re2.h"
+
+const char kTestHost[] = "a.test.com";
+const char kTestAmpPage[] = "/test.html";
+const char kTestSimpleNonAmpPage[] = "/simple.html";
+const char kTestCanonicalPage[] = "/simple_canonical.html";
+const char kTestBody[] =
+    R"(
+      <html amp>
+    <head>
+    <link rel='canonical'
+    href='https://%s:%s%s'>
+    </head>
+    </html>
+    )";
+class DeAmpBrowserTest : public InProcessBrowserTest {
+ public:
+  DeAmpBrowserTest() {
+    feature_list_.InitAndEnableFeature(de_amp::features::kBraveDeAMP);
```

Comment:
we should also verify that we don't redirect when the feature is disabled - can be follow up